### DISCUSSION TITLE
fastMRI sample in SSIM loss

### DISCUSSION THREAD

**Discussion by Khoa-NT** [Project-MONAI/MONAI#4703] [Q&A] [answered]
Hi, thank you for adding the [SSIM loss](https://docs.monai.io/en/latest/losses.html#ssimloss).
Can I ask one question for this term: 

> A fastMRI sample should use the 2D format with C being the number of slices.

In my knownledge, you mean reshape the 3D volume from `(B,C,W,H,D)` to `(B,C*D,W,H)` ?
However, the current [code](https://github.com/Project-MONAI/MONAI/blob/dev/monai/losses/ssim_loss.py#L48) only support C=1, therefore, the loss can't run with `(B,C*D,W,H)`.

Can someone explain this to me? @mersad95zd

**Comment by mersad95zd** [comment, marked as answer]
Hi @Khoa-NT, thanks for your question.

> > A fastMRI sample should use the 2D format with C being the number of slices. 

> In my knownledge, you mean reshape the 3D volume from `(B,C,W,H,D)` to `(B,C*D,W,H)` ?

If you're working on the fastMRI dataset, then no need to reshape because fastMRI's default shape is `(B,num_slices,W,H)`.

> However, the current [code](https://github.com/Project-MONAI/MONAI/blob/dev/monai/losses/ssim_loss.py#L48) only support C=1, therefore, the loss can't run with `(B,C*D,W,H)`.

Good point, since there are datasets where `C` varies across samples, the current implementation works with `C=1` and the loss function should be called for each channel, separately. We'll correct this in the docstring.

Does this address your issue?

**Comment by a-parida12** [comment]
I have a 2D image of batch size =128 and channel 3. What is the correct way to use the SSIMLOss

```python
from monai.losses.ssim_loss import SSIMLoss
criterion = SSIMLoss(spatial_dims=2)
loss = criterion(images, pred, data_range=images.max().unsqueeze(0))
# images.shape == pred.shape == torch.Size([128, 3, 64, 64])
```
In the above code, I get a `RuntimeError`.

`RuntimeError: shape '[1]' is invalid for input of size 3`
